## How to manage data with pandas library

### Reading information froma csv file

In this example I want to show you how to use ***pandas library*** to read data from an external resource as a **csv** file.
For this example we are going to read a file with the information about different apartament offers in Cali city (name, area, address, price, etc).


In [23]:
#Import pandas library
import pandas as pd

#Define a variable to save our data and read the file using the read_csv function
original_data = pd.read_csv("./01_apartments_information.csv")

After reading our file with the information we can use the ***shape*** attribute to know about our data file.

In [16]:
original_data.shape

(1755, 11)

We saw that our file has information about **1755** records and **11** columns. Now we can examine the content of our DataFrame using the **head()** function that can show us by default the first five rows of our data, also we can define as a parameter the number of rows that we want to see ***head(# rows)***

In [21]:
original_data.head()

Proyecto                Vende           Dirección  \
0            Ed. Rincón del Bosque           Inv. CIMAR   AV. 10 N # 51N-36   
1  Vertice Suites - Apartaestudios  C Y C Urbanizadores  Cr. 83E Cl. 48 Bis   
2                          Trikala      Const. Sintagma     Cr. 49 # 14C-15   
3            Edificio Camp. Towers     Hérnandez Bohmer     Cr. 100 Cl. 11A   
4  Vertice Suites - Apartaestudios  C Y C Urbanizadores  Cr. 83E Cl. 48 Bis   

    Tipo   Zona  Alc. # Gar.  Area       Sub Zona Piso      Psep13  
0  Apto.  Norte     1      0  20.0       La Flora  101  28,000,000  
1  Apto.    Sur     1      0  28.0          Caney  303  42,500,000  
2  Apto.    Sur     2   2 a1  29.0        Ingenio  301  41,300,000  
3  Apto.    Sur     1      0  29.0  Ciudad Jardín    7  53,753,490  
4  Apto.    Sur     1      0  29.0          Caney  309  48,500,000

As we can see in our data set, pandas created a column with a index, now we can make that pandas use this column for the index with the **index_col** parameter:

In [22]:
original_data = pd.read_csv("./01_apartments_information.csv", index_col =0)
original_data.head()

Vende           Dirección  \
Proyecto                                                                   
Ed. Rincón del Bosque                     Inv. CIMAR   AV. 10 N # 51N-36   
Vertice Suites - Apartaestudios  C Y C Urbanizadores  Cr. 83E Cl. 48 Bis   
Trikala                              Const. Sintagma     Cr. 49 # 14C-15   
Edificio Camp. Towers               Hérnandez Bohmer     Cr. 100 Cl. 11A   
Vertice Suites - Apartaestudios  C Y C Urbanizadores  Cr. 83E Cl. 48 Bis   

                                  Tipo   Zona  Alc. # Gar.  Area  \
Proyecto                                                           
Ed. Rincón del Bosque            Apto.  Norte     1      0  20.0   
Vertice Suites - Apartaestudios  Apto.    Sur     1      0  28.0   
Trikala                          Apto.    Sur     2   2 a1  29.0   
Edificio Camp. Towers            Apto.    Sur     1      0  29.0   
Vertice Suites - Apartaestudios  Apto.    Sur     1      0  29.0   

                                      Sub Zona Piso      Psep13  
Proyecto                                                         
Ed. Rincón del Bosque                 La Flora  101  28,000,000  
Vertice Suites - Apartaestudios          Caney  303  42,500,000  
Trikala                                Ingenio  301  41,300,000  
Edificio Camp. Towers            Ciudad Jardín    7  53,753,490  
Vertice Suites - Apartaestudios          Caney  309  48,500,000